generators/client/templates/angular/src/main/webapp/app/shared/directive/_sort-by.directive.ts

```diff
@@ -15,7 +14,27 @@ export class <%=jhiPrefixCapitalized%>SortByDirective {
         this.<%=jhiPrefix%>Sort = <%=jhiPrefix%>Sort;
     }
 
-    onClick() {
-        this.jhSort.sort(this.jhSortBy);
+    @HostListener('click') onClick() {
```

Comment:
I worked on the sort directive, it was in my initial branch. It was working.  I looked at @sendilkumarn version and it is not apparent to me how it can work without emitting state change parameters to the host before calling the state change callback. I have not tested the soultion myself, just my quck thoughts after reviewing the changes.

Also, the selector was changed, but not all the places where it was used were.